2021/TurcoZiccolella/app.py

```diff
@@ -0,0 +1,491 @@
+
+                       'DS:MeanIn:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:MeanOut:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:MeanBoth:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+
+                       'DS:VarianceIn:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:VarianceOut:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:VarianceBoth:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+
+                       'DS:StdDevIn:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:StdDevOut:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:StdDevBoth:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+
+                       'DS:UpperBoundIn:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:LowerBoundIn:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:UpperBoundOut:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:LowerBoundOut:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:UpperBoundBoth:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value),
+                       'DS:LowerBoundBoth:GAUGE:{}:{}:{}'.format(hb, min_rrd_value, max_rrd_value)
+                       )
+
+        first_cycle = True
+        ntalkers = 0 
+        i = 0 #Cycle counter
+
+        while True:
+
+            # ---------------------------------------------------------#
+            #          FIRST PHASE: Preparing data from pcap          #
+            # ---------------------------------------------------------#
+
+            # Reset Period/Cycle variables
+            if first_cycle or i == rk_tim:
+                i = 0
+                period_in = 0
+                period_out = 0
+                period_both = 0
+
+                for talker in talkers.values():
+                    talker.ResetTotalPeriodCounter()
+
+                first_cycle = False
+
+            cycle_in_sum = 0
+            cycle_out_sum = 0
+            cycle_both_sum = 0
+
+            # Reconstruct the file name
+            ts,f_name= pl.get_message() #Blocking 
+            timestring = datetime.datetime.fromtimestamp(ts).strftime(p)
+            print("nuovo file generato , aggiorno\n\n\n")
+
+            os.seteuid(0) #Enabling root -> we are reading
+
+            flow_streamer = NFStreamer(source=f_name, statistical_analysis=True, promiscuous_mode=True,
+                                       accounting_mode=0)
+
+            flows = list(flow_streamer) #temporary structure to avoid to hold root 
+            
+            os.remove(f_name)  # Remove temporary file 
+            os.seteuid(int(os.environ.get('SUDO_GID'))) #Disabling root
```

Comment:
Nel processo principale si hanno permessi di root soltanto in questa porzione di codice, dove si accede al file temporaneo creato dal produttore, vengono copiati in una lista di flussi da flowstreamer, si cambia da root  a non root e viceversa settando gli effective user id